crates/db_views/src/post/post_view.rs

```diff
@@ -460,6 +459,22 @@ impl<'a> PostQuery<'a> {
       }
 
       query = query.filter(filter_blocked());
+      if let Some(local_user_id) = my_local_user_id {
+        let blocked_keywords: Vec<String> =
+          LocalUserKeywordBlock::read(pool, local_user_id).await?;
+        if !blocked_keywords.is_empty() {
+          for keyword in blocked_keywords {
+            let pattern = format!("%{}%", keyword);
+            query = query.filter(post::name.not_ilike(pattern.clone()));
+            query = query.filter(post::url.is_null().or(post::url.not_ilike(pattern.clone())));
+            query = query.filter(
+              post::body
+                .is_null()
+                .or(post::body.not_ilike(pattern.clone())),
+            );
+          }
```

Comment:
The performance of this I'm sure is going to be awful, but we'll likely find that out later.